celery/worker/request.py

```diff
@@ -627,15 +628,18 @@ def group_index(self):
         return self._request_dict.get('group_index')
 
 
-def create_request_cls(base, task, pool, hostname, eventer,
+def create_request_cls(app, base, task, pool, hostname, eventer,
```

Comment:
Do we trust this to not break things in downstream/user code? If not, could we make this a kwarg and maybe default to `current_app` if it's not specified?